benchmarks/fast_sin_cos_2π_benchmark.cpp

```diff
@@ -7,23 +7,54 @@
 #include <vector>
 
 #include "benchmark/benchmark.h"
+#include "quantities/numbers.hpp"
 
 namespace principia {
 namespace numerics {
+namespace {
+
+static const __m128d mantissa_bits =
+    _mm_castsi128_pd(_mm_cvtsi64_si128(0x000F'FFFF'FFFF'FFFF));
+// When iterated, the quadrant of the result is unbiased; the latency is
+// between 1 and 2 cycles: at worst this is an and and an xor, at best the xor
+// can only be computed given both trigonometric lines.
+double MixTrigonometricLines(double cos_2πx, double sin_2πx) {
+  __m128d const cos_mantissa_bits =
+      _mm_and_pd(_mm_set_sd(cos_2πx), mantissa_bits);
+  __m128d const sin_all_bits = _mm_set_sd(sin_2πx);
+  __m128d const mixed_bits = _mm_xor_pd(cos_mantissa_bits, sin_all_bits);
+  return _mm_cvtsd_f64(mixed_bits);
+}
+static const __m128d mantissa_bits_and_4_bits_of_exponent =
+    _mm_castsi128_pd(_mm_cvtsi64_si128(0x00FF'FFFF'FFFF'FFFF));
+// Same as above, but when iterated, the result is quickly confined to the
+// principal quadrant.
+double PoorlyMixTrigonometricLines(double cos_2πx, double sin_2πx) {
```

Comment:
Could we have a single function and pass in the mask?  It's not so nice to duplicate fairly complicated intrinsics code.